File: mikado-method/test-data-builders-kata/typescript/test/com/murex/tbw/report/AuthorBuilder.ts
import { Author } from "../../../../../src/com/murex/tbw/domain/book";
import { Country } from "../../../../../src/com/murex/tbw/domain/country";
import { CountryBuilder } from "./CountryBuilder";

export class AuthorBuilder {
    private name: string = "John Doe";
    private nationality: CountryBuilder = CountryBuilder.aCountry();
    static anAuthor(): AuthorBuilder {
        return new AuthorBuilder();
    }


    named(name: string): AuthorBuilder {
        this.name = name;
        return this;
    }


    bornIn(country: CountryBuilder): AuthorBuilder {
        this.nationality = country;
        return this;
    }

    build(): Author {
        return new Author(this.name, this.nationality.build());
    }
}

export const anAuthor = (): AuthorBuilder => AuthorBuilder.anAuthor();

File: mikado-method/test-data-builders-kata/csharp/BookInvoicing.Tests/AuthorBuilder.cs
using BookInvoicing.Domain.Book;
using BookInvoicing.Domain.Country;
using System;

namespace BookInvoicing.Tests
{
    public class AuthorBuilder 
    {
        private string _name = "Bob";
        private Country _country = CountryBuilder.Usa().Build();
        private CountryBuilder _countryBuilder;

        public static AuthorBuilder AnAuthor()
        {
            return new AuthorBuilder();
        }

        public AuthorBuilder Named(string name)
        {
            _name = name;
            return this;
        }

        public AuthorBuilder BornIn(Country country)
        {
            _country = country;
            return this;
        }

        public Author Build()
        {
            return new Author(_name, _countryBuilder is null ? _country : _countryBuilder.Build());
        }

        internal AuthorBuilder BornIn(CountryBuilder country)
        {
            _countryBuilder = country;
            return this;
        }
    }
}